torch/fx/experimental/sym_node.py

```diff
@@ -395,6 +395,28 @@ def expect_size(self, file, line):
             _advise_is_size(SymInt(self))
         return r
 
+    def guard_size_oblivious(self, file, line):
+        """
+        Like guard_bool, but if we encounter unbacked symbols, if those symbols
+        are size-like, we will treat them as >= 2 for the purposes of the analysis.
+
+        This CHANGES the runtime semantics, but all size-oblivious sites have been
+        audited to ensure that the runtime semantics don't change in a material way.
+        Acceptable runtime semantic changes are, e.g., squeeze() no longer dropping
+        an unbacked one size, or a tensor reporting as non-contiguous even if it's
+        contiguous when it is one size.
+        """
+        # TODO: use the file/line for some useful diagnostic on why a
+        # guard occurred
```

Comment:
maybe have file and line optional and use traceback if they're not passed in ?

Reply:
This is a biiiig preexisting condition, I need to figure out something for it, not in this PR though :P